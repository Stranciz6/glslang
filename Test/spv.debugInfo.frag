#version 450

struct S {
    int a;
};

uniform ubuf {
    S s;
};

layout(location = 0) in vec4 inv;
layout(location = 0) out vec4 outv;

void foo(S s)
{
    outv = s.a * inv;
}

void main()
{
    foo(s);
}