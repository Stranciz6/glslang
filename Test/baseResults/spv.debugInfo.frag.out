spv.debugInfo.frag
Warning, version 450 is not yet complete; most version-specific features are present, but some are missing.

// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 40

                              Capability Shader
               2:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Fragment 5  "main" 17 24
                              ExecutionMode 5 OriginUpperLeft
               1:             String  "spv.debugInfo.frag"
                              Source GLSL 450 1  "#version 450

struct S {
    int a;
};

uniform ubuf {
    S s;
};

layout(location = 0) in vec4 inv;
layout(location = 0) out vec4 outv;

void foo(S s)
{
    outv = s.a * inv;
}

void main()
{
    foo(s);
}"
                              Name 5  "main"
                              Name 8  "S"
                              MemberName 8(S) 0  "a"
                              Name 12  "foo(struct-S-i11;"
                              Name 11  "s"
                              Name 17  "outv"
                              Name 24  "inv"
                              Name 27  "S"
                              MemberName 27(S) 0  "a"
                              Name 28  "ubuf"
                              MemberName 28(ubuf) 0  "s"
                              Name 30  ""
                              Name 31  "S"
                              MemberName 31(S) 0  "a"
                              Name 33  "param"
                              Decorate 17(outv) Location 0
                              Decorate 24(inv) Location 0
                              MemberDecorate 27(S) 0 Offset 0
                              MemberDecorate 28(ubuf) 0 Offset 0
                              Decorate 28(ubuf) Block
                              Decorate 30 DescriptorSet 0
               3:             TypeVoid
               4:             TypeFunction 3
               7:             TypeInt 32 1
            8(S):             TypeStruct 7(int)
               9:             TypePointer Function 8(S)
              10:             TypeFunction 3 9(ptr)
              14:             TypeFloat 32
              15:             TypeVector 14(float) 4
              16:             TypePointer Output 15(fvec4)
        17(outv):     16(ptr) Variable Output
              18:      7(int) Constant 0
              19:             TypePointer Function 7(int)
              23:             TypePointer Input 15(fvec4)
         24(inv):     23(ptr) Variable Input
           27(S):             TypeStruct 7(int)
        28(ubuf):             TypeStruct 27(S)
              29:             TypePointer Uniform 28(ubuf)
              30:     29(ptr) Variable Uniform
           31(S):             TypeStruct 7(int)
              32:             TypePointer Function 31(S)
              34:             TypePointer Uniform 27(S)
         5(main):           3 Function None 4
               6:             Label
       33(param):     32(ptr) Variable Function
              35:     34(ptr) AccessChain 30 18
              36:       27(S) Load 35
              37:      7(int) CompositeExtract 36 0
              38:     19(ptr) AccessChain 33(param) 18
                              Store 38 37
              39:           3 FunctionCall 12(foo(struct-S-i11;) 33(param)
                              Return
                              FunctionEnd
12(foo(struct-S-i11;):           3 Function None 10
           11(s):      9(ptr) FunctionParameter
              13:             Label
              20:     19(ptr) AccessChain 11(s) 18
              21:      7(int) Load 20
              22:   14(float) ConvertSToF 21
              25:   15(fvec4) Load 24(inv)
              26:   15(fvec4) VectorTimesScalar 25 22
                              Store 17(outv) 26
                              Return
                              FunctionEnd
